lib/feedx/pusher.rb

```diff
@@ -4,13 +4,22 @@
 module Feedx
   # Pushes a relation as a protobuf encoded stream to an S3 location.
   class Pusher
-    # @param [Enumerable,ActiveRecord::Relation] relation to stream.
+    # See constructor.
+    def self.perform(url, opts={}, &block)
+      new(url, opts, &block).perform
+    end
+
     # @param [String] url the destination URL.
     # @param [Hash] opts options
+    # @option opts [Enumerable,ActiveRecord::Relation] :enum relation or enumerator to stream.
```

Comment:
IMO, just another option to know/remember about, but doesn't actually add much value. I think accepting block is more than enough.